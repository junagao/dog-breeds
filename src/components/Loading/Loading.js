import React from 'react';

import SpinnerDiv from './Loading.styles';

const Loading = () => <SpinnerDiv data-testid="spinner" />;

export default Loading;
