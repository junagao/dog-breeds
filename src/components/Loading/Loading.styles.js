import styled from 'styled-components';

const SpinnerDiv = styled.div`
  border: 0.25rem solid #f3f3f3;
  border-radius: 50%;
  border-top: 0.25rem solid #777;
  width: 2rem;
  height: 2rem;
  animation: spin 0.75s linear infinite;
  margin: 0 auto;
  margin-top: 2rem;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export default SpinnerDiv;
