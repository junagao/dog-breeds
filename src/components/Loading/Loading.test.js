import React from 'react';
import { render } from '@testing-library/react';
import Loading from './Loading';

describe('Loading', () => {
  test('matches snapshot', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });

  test('should render the loading spinner correctly with one SpinnerDiv element', () => {
    const { getByTestId } = render(<Loading />);
    expect(getByTestId('spinner')).toBeInTheDocument();
  });
});
