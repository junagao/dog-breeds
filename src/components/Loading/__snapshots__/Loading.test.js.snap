// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches snapshot 1`] = `
.c0 {
  border: 0.25rem solid #f3f3f3;
  border-radius: 50%;
  border-top: 0.25rem solid #777;
  width: 2rem;
  height: 2rem;
  -webkit-animation: spin 0.75s linear infinite;
  animation: spin 0.75s linear infinite;
  margin: 0 auto;
  margin-top: 2rem;
}

<div>
  <div
    class="c0"
    data-testid="spinner"
  />
</div>
`;
